src/client.rs

```diff
@@ -395,162 +318,119 @@ pub async fn run_client(
-							.map(|(i, col)| to_column_cells(i as u16, col))
-							.collect::<Vec<Vec<Cell>>>();
-
-						let layout = layout_from_index(
-							block.header.app_data_lookup.index.as_slice(),
-							block.header.app_data_lookup.size,
-						);
-
-						let ext = reconstruct_app_extrinsics(
-							layout,
-							columns,
-							(block.max_rows / 2) as usize,
-							32,
-						);
-						log::debug!("Reconstructed extrinsic: {:?}", ext);
-
-						match construct_matrix(
-							block.num,
-							block.max_rows,
-							block.max_cols,
-							arced_cells,
-						) {
-							Ok(matrix) => {
-								match push_matrix(matrix, latest_cid, &ipfs, &pin).await {
-									Ok(cid) => {
-										latest_cid = Some(cid);
-										// publish block-cid mapping message over gossipsub network
-										match prepare_block_cid_fact_message(block.num as i128, cid)
-										{
-											Ok(msg) => {
-												match ipfs.publish("topic/block_cid_fact", msg) {
-													Ok(_) => {
-														// thread-safely put an entry in local in-memory store
-														// for block to cid mapping
-														//
-														// it can be used later for for serving clients or
-														// answering to questions asked by other peers over
-														// gossipsub network
-														//
-														// once a CID is self-computed, it'll never be rewritten even
-														// when conflicting fact is found over gossipsub channel
-														match set_block_cid_entry(
-															block_cid_store.clone(),
-															block.num as i128,
-															BlockCidPair {
-																cid,
-																self_computed: true, // because this block CID is self-computed !
-															},
-														) {
-															Ok(_) => {},
-															Err(e) => {
-																log::info!("error: {}", e);
-															},
-														};
-														log::info!(
-															"✅ Block {} available\t{}",
-															block.num,
-															cid.clone()
-														);
-													},
-													Err(_) => {
-														log::info!("error: failed to publish fact on `topic/block_cid_fact` topic");
-													},
-												};
-											},
-											Err(msg) => {
-												log::info!("error: {}", msg);
-											},
-										};
-									},
-									Err(msg) => {
-										log::info!("error: {}", msg);
-									},
-								}
-							},
-							Err(msg) => {
-								log::info!("error: {}", msg);
-							},
-						};
-					},
-					Err(e) => {
-						log::info!("error: {}", e);
-					},
-				};
-			},
-			Err(e) => {
-				log::info!("Error encountered while listening for blocks: {}", e);
-				break;
-			},
-		};
+		let rpc_ = check_http(&full_node_rpc).await?;
+		let block_cid_entry = get_block_cid_entry(store, block.num as i128).map(|pair| pair.cid);
+
+		let ipfs_cells = get_matrix(ipfs, block_cid_entry)
+			.inspect_err(|err| log::error!("Fail to fetch cells from IPFS: {}", err))
```

Comment:
We don't make distinction between BlockNotFound and other error causes, which is maybe fine, but it can raise false alarms if log level is error. We should handle it as specific case or just set it to info? Or just put TODO and handle it later on all places where ipfs.get is called?